src/components/comparison/layers-panel/layers-control-panel.tsx

```diff
@@ -70,23 +72,65 @@ export const LayersControlPanel = ({
   const [showLayerSettings, setShowLayerSettings] = useState<boolean>(false);
   const [layerToDeleteId, setLayerToDeleteId] = useState<string>("");
 
-  const handleSelectLayer = (layerId: string, parentId: string) => {
-    onLayerSelect(layerId, parentId);
+  const handleSelectAllLeafsInGroup = (
```

Comment:
`getAllLayersInGroupsTree` in `comparison-side.tsx` has same logic. I suggest creating `layer-utils` for it.